torch/_inductor/analysis.py

```diff
@@ -0,0 +1,24 @@
+from torch._inductor import scheduler
+
+from . import ir
+
+def get_runtime_of_snode(snode: "BaseSchedulerNode"):
+    """
+    Gets the runtime of Scheduler node. Currently somewhat of a placeholder, todo to be replaced by more sophisticated approaches.
+
+    TODO(yf225): potentially replace with a more sophisticated cost model (e.g. analytical)
+    """
+    # print(f"get_runtime_of_snode: snode.node: {snode.node}")
+    if isinstance(snode.node, ir.AllReduce):
```

Comment:
another q: do you plan to add `allgather` and `reduce_scatter` in follow up PRs? asking bc in training ppl mostly use sequence parallel and it would be great if we can add these two :)

Reply:
all other collectives will go to the `ir.CollectiveKernel` branch, also we desperately need a better cost model for this..